### ISSUE TITLE
[Bug]: Milvus crush after running node-sdk test-case for about 50 times

### ISSUE THREAD

**Issue by shanghaikid** [milvus-io/milvus#24458]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-latest
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): node-sdk main branch
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

crushed

### Milvus Log

```
[2023/05/27 12:28:28.447 +00:00] [INFO] [proxy/impl.go:1016] ["GetCollectionStatistics enqueued"] [traceID=36db1e25924728c] [role=proxy] [msgID=441762580635058178] [BeginTS=441762580635058178] [EndTS=441762580635058178] [db=Collection] [collection=collection_xi1ixmum]
[2023/05/27 12:28:28.488 +00:00] [DEBUG] [datanode/flow_graph_dd_node.go:348] ["DDNode sent delta timeTick"] [collectionID=441756972419693190] [ts=441762580619853829] [ts_p=2023/05/27 12:28:28.346 +00:00] [channel=by-dev-rootcoord-dml_6_441756972419693190v0]
[2023/05/27 12:28:28.509 +00:00] [DEBUG] [datanode/flow_graph_dd_node.go:210] ["DDNode receive insert messages"] [numRows=1] [vChannelName=by-dev-rootcoord-dml_15_441756972419693174v0]
[2023/05/27 12:28:28.512 +00:00] [INFO] [datanode/channel_meta.go:217] ["adding segment"] [type=New] [segmentID=441756972419693232] [collectionID=441756972419693174] [partitionID=441756972419693175] [channel=by-dev-rootcoord-dml_15_441756972419693174v0] [startPosition="channel_name:\"by-dev-rootcoord-dml_15_441756972419693174v0\" msgID:\"\\007\\323\\3711\\240o!\\006\" msgGroup:\"by-dev-dataNode-8-by-dev-rootcoord-dml_15_441756972419693174v0\" timestamp:441762580547239938 "] [endPosition="channel_name:\"by-dev-rootcoord-dml_15_441756972419693174v0\" msgID:\"W\\323\\3711\\240o!\\006\" msgGroup:\"by-dev-dataNode-8-by-dev-rootcoord-dml_15_441756972419693174v0\" timestamp:441762580607533057 "] [recoverTs=0] [importing=false]
[2023/05/27 12:28:28.512 +00:00] [INFO] [datanode/channel_meta.go:374] ["begin to init pk bloom filter"] [segmentID=441756972419693232] [statsBinLogsLen=0]
[2023/05/27 12:28:28.512 +00:00] [WARN] [datanode/channel_meta.go:403] ["no stats files to load"] [segmentID=441756972419693232]
[2023/05/27 12:28:28.512 +00:00] [INFO] [datanode/channel_meta.go:561] ["updating segment num row"] ["Segment ID"=441756972419693232] [numRows=1]
[2023/05/27 12:28:28.513 +00:00] [INFO] [datanode/channel_meta.go:576] ["updating segment memorySize"] ["Segment ID"=441756972419693232] [memorySize=61]
[2023/05/27 12:28:28.513 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:289] ["segment buffer status"] [no.=0] [segmentID=441756972419693232] [channel=by-dev-rootcoord-dml_15_441756972419693174v0] [size=1] [limit=294338] [memorySize=61]
[2023/05/27 12:28:28.535 +00:00] [WARN] [datanode/flow_graph_dd_node.go:185] ["filter invalid insert message, collection mis-match"] ["Get collID"=441756972419693174] ["Expected collID"=441756972419693066]
fatal error: unexpected signal during runtime execution
[2023/05/27 12:28:31.446 +00:00] [INFO] [observers/collection_observer.go:177] ["collection targets"] [collectionID=441756972419693060] [segmentTargetNum=0] [channelTargetNum=1] [totalTargetNum=1] [replicaNum=1]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x2 pc=0x4002d52ec5]

runtime stack:
runtime.throw({0x4064ffd?, 0x4055e74390?})
	/usr/local/go/src/runtime/panic.go:992 +0x71
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:802 +0x3a9

goroutine 852654 [syscall]:
runtime.cgocall(0x31354f0, 0xc0028075f8)
	/usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc0028075d0 sp=0xc002807598 pc=0x147857c
github.com/milvus-io/milvus/internal/querynode._Cfunc_LoadFieldData(0x4056f6c700, 0x40565810d0)
	_cgo_gotypes.go:692 +0x55 fp=0xc0028075f8 sp=0xc0028075d0 pc=0x2dc43b5
github.com/milvus-io/milvus/internal/querynode.(*Segment).LoadMultiFieldData.func1(0x14c0bd4?, 0xc0010004e0?)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:867 +0x65 fp=0xc002807638 sp=0xc0028075f8 pc=0x2de4045
github.com/milvus-io/milvus/internal/querynode.(*Segment).LoadMultiFieldData(0xc0023a7b00, 0x96, {0xc003b79200, 0x7, 0xc000202900?})
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:867 +0x2f5 fp=0xc0028077a8 sp=0xc002807638 pc=0x2de3c75
github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadFiles(0xc001d2f320, {0x44a88b8, 0xc002e22b10}, 0xc0023a7b00, 0xc002131770)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:313 +0x4a5 fp=0xc002807be0 sp=0xc0028077a8 pc=0x2d8f1e5
github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment.func3(0xc000060500?)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:203 +0xe5 fp=0xc002807dc0 sp=0xc002807be0 pc=0x2d8e785
github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func3(0x0)
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:83 +0x16d fp=0xc002807fc8 sp=0xc002807dc0 pc=0x20a24ad
github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func4()
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:101 +0x2a fp=0xc002807fe0 sp=0xc002807fc8 pc=0x20a230a
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc002807fe8 sp=0xc002807fe0 pc=0x14e1c81
created by github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:70 +0x1aa

```
[milvus.log.zip](https://github.com/milvus-io/milvus/files/11581893/milvus.log.zip)

**Comment by shanghaikid** [comment]
restarted the docker, the milvus worked again.

**Comment by xiaofan-luan** [comment]
@shanghaikid 
what is the test case does?
looks like there is insert and load. Anything else?

**Comment by shanghaikid** [comment]
> @shanghaikid what is the test case does? looks like there is insert and load. Anything else?

Actually I meet this error for about 2 times last week. 

In most features, we do a fully test to make sure the feature is working as expected, for example, the partition key feature, we will do
1. create a collection
2. insert some data
3. create index
4. load 
5. search or query

we use jest, a js framework that can run tests concurrently.   not sure if it is a problem.

By the way, we only insert 20 - 150 entities per test. 


I guess,  it is due to the json feature. I meet this crash when I was implementing the json feature.

**Comment by shanghaikid** [comment]
Another observation: because some of my test cases will invoke releaseCollection, in the version, the ReleaseCollection function is **very very very** slow, sometimes will timeout. 

```
  console.info
    [DB:Collection:GetLoadingProgress] executed in 8ms.

      at Object.onReceiveStatus (milvus/utils/Grpc.ts:191:25)

  console.info
    [DB:Collection:GetLoadingProgress] executed in 6ms.

      at Object.onReceiveStatus (milvus/utils/Grpc.ts:191:25)

  console.info
    [DB:Collection:GetLoadingProgress] executed in 5ms.

      at Object.onReceiveStatus (milvus/utils/Grpc.ts:191:25)

  console.info
    [DB:Collection:GetLoadState] executed in 8ms.

      at Object.onReceiveStatus (milvus/utils/Grpc.ts:191:25)

  console.info
    [DB:Collection:ReleaseCollection] is timeout, timeout set: 10000ms.

      at Object.onReceiveStatus (milvus/utils/Grpc.ts:164:35)

  console.info
    [DB:Collection:ShowCollections] executed in 8ms.
```

**Comment by yah01** [comment]
The crash problem may be a Knowhere bug, the knowhere team is working on it.

The slow release problem is fixed with https://github.com/milvus-io/milvus/pull/24523

**Comment by shanghaikid** [comment]
> The crash problem may be a Knowhere bug, the knowhere team is working on it.
> 
> The slow release problem is fixed with #24523

Yes, the slow release problem is fixed with the latest image. thanks.

**Comment by yah01** [comment]
/assign @shanghaikid 
The crash problem fixed with #24567

**Comment by yah01** [comment]
see related issue #24566
